app/services/salesman/update.rb

```diff
@@ -0,0 +1,38 @@
+module Services
+  module Salesman
+    class Update
+      def initialize(id, name, status)
+        @name = name
+        @status = status
+        @id = id
+      end
+
+      def call
+        update_salesman
+      end
+
+      private
+
+      attr_reader :name, :status, :id
+
+      def update_salesman
+        salesman = ::Salesman.find_by(id: id)
+        if salesman.blank?
+          404
```

Comment:
pass the status to the controller. Here you should think generically and can only send nil.